app/src/test/java/com/mho/training/TestDI.kt

```diff
@@ -0,0 +1,153 @@
+package com.mho.training
+
+import com.example.android.data.repositories.*
+import com.example.android.data.sources.*
+import com.example.android.domain.*
+import com.example.android.domain.result.DataResult
+import com.example.android.testshared.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+import org.koin.core.context.startKoin
+import org.koin.core.module.Module
+import org.koin.dsl.module
+
+fun initMockedDi(vararg modules: Module) {
+    startKoin {
+        modules(listOf(mockedAppModule, mockedDataModule) + modules)
+    }
+}
+
+private val mockedAppModule = module {
```

Comment:
You could create a dataSourceModule in order to separate the concerns.

Reply:
I'll check it.